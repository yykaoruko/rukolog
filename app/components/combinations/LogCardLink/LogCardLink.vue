<template>
  <nuxt-link :to="to" class="log-card-link">
    <div class="log-card-link__wrap">
      <log-title class="log-card-link__wrap__title" :title="title" />
      <app-date class="log-card-link__wrap__date" :datetime="datetime" />
      <log-description
        class="log-card-link__wrap__description"
        :text="description"
      />
    </div>
  </nuxt-link>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import AppDate from '@/components/basics/AppDate/AppDate.vue';
import LogDescription from '@/components/basics/LogDescription/LogDescription.vue';
import LogTitle from '@/components/basics/LogTitle/LogTitle.vue';
export default Vue.extend({
  name: 'LogCardLink',
  components: {
    AppDate,
    LogDescription,
    LogTitle,
  },
  props: {
    datetime: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      required: true,
    },
    to: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="scss" scoped>
.log-card-link {
  display: block;
  border-left: 4px solid $color-sub;
  text-decoration: none;
}
.log-card-link__wrap {
  padding: 16px 24px;
  transition: all 0.2s;
  background: $color-base;
  &:hover {
    background: $color-accent;
  }
}
.log-card-link__wrap__title {
  color: $color-text;
}
.log-card-link__wrap__date {
  color: $color-text;
}
.log-card-link__wrap__description {
  display: none;
}
</style>
