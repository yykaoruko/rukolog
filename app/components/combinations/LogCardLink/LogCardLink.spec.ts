import Vue from 'vue';
import { shallowMount, Wrapper, RouterLinkStub } from '@vue/test-utils';
import LogCardLink from './LogCardLink.vue';

describe('LogCardLink', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LogCardLink, {
        propsData: {
          datetime: '2019-10-07',
          title: 'title',
          to: 'to',
          tags: ['hoge'],
        },
        stubs: {
          NuxtLink: RouterLinkStub,
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
