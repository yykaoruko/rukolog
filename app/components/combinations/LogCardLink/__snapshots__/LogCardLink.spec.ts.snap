// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogCardLink mount snapshot 1`] = `
<a
  class="log-card-link"
  tags="hoge"
>
  <div
    class="log-card-link__wrap"
  >
    <log-title-stub
      class="log-card-link__wrap__title"
      title="title"
    />
     
    <app-date-stub
      class="log-card-link__wrap__date"
      datetime="2019-10-07"
    />
     
    <log-description-stub
      class="log-card-link__wrap__description"
      text=""
    />
  </div>
</a>
`;
